torch/testing/_asserts.py

```diff
@@ -157,7 +228,13 @@ def _check_attributes_equal(
     if check_dtype and actual.dtype != expected.dtype:
         return AssertionError(msg_fmtstr.format("dtype", actual.dtype, expected.dtype))
 
-    if check_stride and actual.stride() != expected.stride():
+    if actual.is_sparse != expected.is_sparse:
+        return AssertionError(msg_fmtstr.format("is_sparse", actual.is_sparse, expected.is_sparse))
+    elif actual.is_sparse and check_is_coalesced and actual.is_coalesced() != expected.is_coalesced():
+        return AssertionError(msg_fmtstr.format("is_coalesced()", actual.is_coalesced(), expected.is_coalesced()))
+    elif actual.is_sparse_csr != expected.is_sparse_csr:
+        return AssertionError(msg_fmtstr.format("is_sparse_csr", actual.is_sparse_csr, expected.is_sparse_csr))
+    elif not (actual.is_sparse or actual.is_sparse_csr) and check_stride and actual.stride() != expected.stride():
```

Comment:
I think this would be more readable if sparse logic were separated from strided logic